### PR TITLE
Adds SRIOV Network Device Plugin integration tests

### PR DESCRIPTION
**Pull request by claudiubelu**
SRIOV Network Device Plugin depends on Multus, which means that the integration test is deploying Multus as well.

### CODE REVIEW

File: tests/integration/test_sriov_net_device_plugin.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        ["k8s", "kubectl", "apply", "-f", "-"],
+        input=daemonset.encode(),
+    )
+
+
+def test_integration_sriov_ndp(tmp_path: Path, module_instance: harness.Instance):
+    # We also need multus in order to test out sriov-network-device-plugin.
+    # It should become Available if everything is fine with it.
+    helm_cmd = k8s_util.get_helm_install_command(
+        "multus", "oci://registry-1.docker.io/bitnamicharts/multus-cni"
```
Comment (bschimke95) [inline tests/integration/test_sriov_net_device_plugin.py:66]: nit: should we use our Multus Rock here?

File: tests/integration/test_sriov_net_device_plugin.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    )
+    module_instance.exec(helm_cmd)
+    k8s_util.wait_for_daemonset(module_instance, "multus-multus-cni", "kube-system")
+
+    _deploy_sriov_ndp(tmp_path, module_instance)
+    k8s_util.wait_for_daemonset(
+        module_instance, "kube-sriov-device-plugin-amd64", "kube-system"
```
Comment (bschimke95) [inline tests/integration/test_sriov_net_device_plugin.py:73]: I guess this will break on ARM, no?

**Review by bschimke95 — COMMENTED** [review]
Looks good, minor comments and clarifications.

File: tests/integration/test_sriov_net_device_plugin.py (hunk 1 above)
Comment (claudiubelu) [reply tests/integration/test_sriov_net_device_plugin.py:66]: Might be an idea, but I would prefer to keep the original image, as a countermeasure if something changed in multus and its CRDs on which this depends on.

File: tests/integration/test_sriov_net_device_plugin.py (hunk 2 above)
Comment (claudiubelu) [reply tests/integration/test_sriov_net_device_plugin.py:73]: We are testing amd64 anyways, as it is the case on all integration tests on all rocks. That is something that can be addressed further down the line, if we will be able to test arm64 rocks, and want to test arm64 rocks. We can address this issue for all rocks at a later date.

File: tests/integration/test_sriov_net_device_plugin.py (hunk 1 above)
Comment (bschimke95) [reply tests/integration/test_sriov_net_device_plugin.py:66]: Fair point

File: tests/integration/test_sriov_net_device_plugin.py (hunk 2 above)
Comment (bschimke95) [reply tests/integration/test_sriov_net_device_plugin.py:73]: alright

**Pull request merged**